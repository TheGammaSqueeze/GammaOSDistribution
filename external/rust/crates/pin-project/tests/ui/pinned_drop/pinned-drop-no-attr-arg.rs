use pin_project::{pin_project, pinned_drop};
use std::pin::Pin;

#[pin_project]
struct S {
    #[pin]
    f: u8,
}

#[pinned_drop]
impl PinnedDrop for S {
    //~^ ERROR E0119
    fn drop(self: Pin<&mut Self>) {}
}

fn main() {}
