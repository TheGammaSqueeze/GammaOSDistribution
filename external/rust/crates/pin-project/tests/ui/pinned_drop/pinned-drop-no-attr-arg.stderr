error[E0119]: conflicting implementations of trait `pin_project::__private::PinnedDrop` for type `S`:
  --> $DIR/pinned-drop-no-attr-arg.rs:11:1
   |
4  | #[pin_project]
   | -------------- first implementation here
...
11 | impl PinnedDrop for S {
   | ^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `S`
